### PR TITLE
Add configuration for client TLS certificate.

### LINKED ISSUE #1363: Environment variables for OTLP client TLS key / certificate
**What are you trying to achieve?**

Ability to set client key / certificate for mTLS.

**Additional context.**

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/protocol/exporter.md#configuration-options

We currently only have a CERTIFICATE_FILE environment variable, which is a bit misleading since this only refers to trusted certificates. For mTLS, there would need to be two additional variables, one for the client certificate and one for the client private key. I think we should have three variables

OTEL_EXPORTER_OTLP_TLS_PRIVATE_KEY
OTEL_EXPORTER_OTLP_TLS_CERTIFICATE
OTEL_EXPORTER_OTLP_TLS_TRUSTED_CERTIFICATE

Reference: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1829

### PR DESCRIPTION
**Pull request by anuraaga**
Fixes #1363

## Changes

Adds OTLP exporter configuration for client TLS to enable mTLS

Related issues https://github.com/open-telemetry/opentelemetry-java/issues/4165

### CODE REVIEW

File: specification/protocol/exporter.md [hunk 1]
```diff
@@ -8,16 +8,18 @@ This document specifies the configuration options available to the OpenTelemetry
...
+| Client Certificate File | The certificae to configure the client's TLS credentials for use with mTLS.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | n/a | `OTEL_EXPORTER_OTLP_CLIENT_PRIVATE_KEY`, `OTLP_EXPORTER_OTLP_TRACES_CLIENT_PRIVATE_KEY`, `OTLP_EXPORTER_OTLP_METRICS_CLIENT_PRIVATE_KEY` |
```
Comment (evantorrie) [inline specification/protocol/exporter.md:18]: s/certificae/certificate/

File: specification/protocol/exporter.md [hunk 2]
```diff
@@ -8,16 +8,18 @@ This document specifies the configuration options available to the OpenTelemetry
...
+| Certificate File        | The trusted certificate to use when verifying a server's TLS credentials. Should only be used for a secure connection.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | n/a               | `OTEL_EXPORTER_OTLP_CERTIFICATE` `OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE` `OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE`                                                                    |
```
Comment (tigrannajaryan) [inline specification/protocol/exporter.md:16]: I know you did not change this, but what is a "trusted certificate"? Is this a certificate that server must present? Or is this a root CA certificate that the server certificate must be signed with? It would be useful to clarify this. Can be a separate PR.

**Comment by tigrannajaryan** [comment]
Github doesn't visualize the diffs on very long lines properly. Just to double check: there are no breaking changes to existing env variable, are there? From what I see this adds 2 new env variables and doesn't change any existing ones. Just wanted to confirm.

**Comment by alolita** [comment]
@tigrannajaryan is the @open-telemetry/technical-committee just routing to you and @evantorrie for spec reviews? The changes don't seem to get merged for many days. Please advise what we could do better.

**Comment by tigrannajaryan** [comment]
> Please advise what we could do better.

Find interested stakeholders, relevant approvers, ask them to review/comment/approve.

On meta level: we need to describe the workflow for PRs (issue [is open](https://github.com/open-telemetry/opentelemetry-specification/issues/1963)), just like I did recently for issues, to make it clear who is responsible for what. My opinion is that the significant burden will be put on the PR author to drive the PR, but I may be wrong.

**Review by tigrannajaryan — APPROVED** [review]
LGTM. Adds 2 env vars to specify (private,public) keys of client-side TLS certificate.

**Comment by tigrannajaryan** [comment]
This really needs a different formatting so that the diffs are useful otherwise reviewing is nearly impossible.

**Comment by yurishkuro** [comment]
> This really needs a different formatting so that the diffs are useful otherwise reviewing is nearly impossible.

+1. The by-language [spec-compliance-matrix.md](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md) makes a lot of sense in a table that has many columns. This table has only 4 columns, and renders very badly anyway because descriptions are long. I would suggest changing to a simpler format:
  * **{option name}**: {description}
    * Default: {value}
    * Env vars: {...}

**Comment by anuraaga** [comment]
Sent #2345 to reformat the table. I'll go ahead and close this PR since it would need to be redone from scratch for that change. Will be good to see the new PR workflow - if the burden is left on the author I'd probably opt-out of contributing to this repo since energy/benefit ratio of PRs would be quite poor.
